### PR TITLE
[WIP]: Add Support for Chrome Extensions; Improve Extensibility Generally  

### PR DESCRIPTION
**Pull request by medelman17**
Will go through this all in depth later on... have to run out, but wanted to gauge interest, get thoughts going, etc.

### CODE REVIEW

**Comment by sw-yx** [comment]
pretty extensive refactor. in future it's best to ask before doing something like this. gonna leave this one to jared

**Comment by medelman17** [comment]
Totally agree; I did :-)

https://twitter.com/jaredpalmer/status/1178079213703909382?s=21

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: src/cli/utils/getTsdxBag.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { getSpinner, getProjectPath, getTemplate, getDeps } from '../utils';
+import { safePackageName, logger } from '../../helpers';
+import { paths, defaultVersion as version } from '../../constants';
+import * as Messages from '../messages';
+
+export async function getTsdxBag(args: { pkg: string; opts: any }) {
```
Comment (jaredpalmer) [inline src/cli/utils/getTsdxBag.ts]: Needs a different name.

**Comment by medelman17** [comment]
Thanks. Yes, I totally understand and appreciate/anticipated feedback re the above. Never really figured this would land out of the box. Also, I’m totally cool with canceling this if way off point. I dunno. 🤷🏻‍♂️ It was a good experience doing it (still not done), and I’m 💯 in learn mode here. No regrets (other than wasting y’all’s time, if that’s the end result).

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sw-yx** [comment]
@medelman17 part of that learning is also how to make others comfortable with your refactor. i dont think either of us are against it at all, it just wasnt a great idea to lump a bunch of different things into one PR. if you do decide to retry, maybe see what's a small yet most impactful extensibility refactor you can propose that you'd like to get feedback on?

**Pull request closed without merging**